statistics-production/api-data-standards.qmd

```diff
@@ -0,0 +1,191 @@
+
+---
+
+## Introduction
+
+The API offers analysts, both internal to the DfE and external consumers and communicators of education statistics,
+a way to programmatically access data on EES. However, in order to ensure a fit for purpose service, not all EES 
+data will be accessible via the API, and any that is will need to pass a higher bar for quality. In effect API data
+**must** meet all the criteria laid out in our [Open data standards guidance](../statistics-production/ud.qmd).
+
+Whilst the EES data screener tests for a significant base level of data quality and consistency, there are some 
+additional criteria that are either too awkward to test for rigorously using the screener or are tested for but
+returned as warnings. Data intended for the EES API must pass all the base level screener tests, plus a number
+that only return warnings, plus manual inspection by the platform gatekeepers. These are primarily:
+
+- Strict tidy data structures - i.e. appropriate use of filters and indicators.
+- Standardised filter col_names and items consistent with the harmonised standards.
+- Standardised indicator col_names meeting the naming standards.
+- Character limits for col_names and filter items.
+
+Examples of these that do and don't meet the API data standards are provided in the following sections.
+
+## Tidy data structure
+
+The key thing on tidy data structure is to avoid filter items being included within indicator col_names. Where 
+you have collections of related terms appearing in indicator names (e.g. male, female, total), then these 
+should be translated into a filter column, with the data being pivoted.
+
+### Example 1 - Attainment rates and scores
+
+#### Example of bad practice
+
+The following would not be accepted for publication via the API.
+
+::: {.table-responsive}
+
+| count_pupils_grade9to5 | count_pupils_grade9to4 | count_pupils_grade9to1 | percent_pupils_grade9to5 | percent_pupils_grade9to4 | percent_pupils_grade9to1 | progress8_score_male | progress8_score_female | progress8_score |attainment8_score_male | attainment8_score_female | attainment8_score |
+|------------------------|------------------------|------------------------|--------------------------|--------------------------|-------------------------|----------------------|------------------------|-----------------|----------------------|------------------------|-----------------|
+| 30                     | 40                     | 50                     | 60                       | 80                       | 100                      | 0.2                  | 0.21                   | 0.21            |0.09                  | 0.08                   | 0.10            |
+
+: Attainment grade rates and scores in non-tidy format
+
+:::
+
+#### Example of good practice
+
+The following would be accepted for publication via the API. In this case, splitting the data into separate data files is required in order to create tidy data structures.
```

Comment:
I guess here they could also use a hierarchy structure here?

Reply:
As opposed to splitting files? In this case, I don't think a hierarchy works in place of splitting the files given the indicators are measuring different things for the different attainment_metrics. Could maybe add another example in of how a hierarchy could work.

Reply:
Fair point, think it might be worth a second example to cover that other case though as some people will have that, and it then gives a nice clear line for when to split files and when not to?

Reply:
Have added something like that as the starting example and shifted the other two along as examples 2 and 3